### DISCUSSION TITLE
Woox R7060 Garden irrigation control

### DISCUSSION THREAD

**Discussion by RenePhilipse** [Koenkk/zigbee2mqtt#9826] [Devices]
Hi All,

I migrated from deconz to zigbee2mqtt since a week or 2 now. So far I am loving it and it is incredible stable! All my devices were recognized out of the box and were announced to Home Assistant without any hussle. So far very positive!

I also have a Woox R7060 irrigation controller, which is recognized and working basically. I can switch it on and off without any problem. The only thing is that I don't see the battery percentage, which is quite handy really. I know that the device is capable of it, because it was showing with Deconz, but in zigbee2mqtt there is only a questionmark showing. The status tab is also not showing anything about batteries:
`{
    "linkquality": 255,
    "state": "OFF"
}`

When I go to the developer console and read the powercfg cluster batteryvoltage and batterypercentage, there is a response:
`debug Received Zigbee message from 'WooxTuinbewatering', type 'readResponse', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":174,"batteryVoltage":9}' from endpoint 1 with groupID null
debug No converter available for 'R7060' with cluster 'genPowerCfg' and type 'readResponse' and data '{"batteryPercentageRemaining":174,"batteryVoltage":9}'
info Read result of 'genPowerCfg': {"batteryVoltage":9,"batteryPercentageRemaining":174}`

I did some digging in the documentation about creating converter files, but is not exactly clear to me. Step 1 starts with the fact that a device is not recognized, but that is not the case here. Only a specific attribute is not reported.

Anyone who can put me in the right direction to solve this? Thanks!

**Comment by RenePhilipse** [comment]
I tried some things with converter files to get the battery status displayed, but no luck so far. Anyone that can help me how to get started?

**Reply by Nerdiyde** [reply]
Hey @RenePhilipse ,
meanwhile I stumbled over the same issue. Did you made any progress on this? :)
Best regards
Fab

**Reply by RenePhilipse** [reply]
Hi @Nerdiyde 

I opened a future request at the creator, and he made the battery percentage available. Currently it is already working in the latest edge (beta) version and i think it will be released also in the next stable version.

**Reply by Nerdiyde** [reply]
Oh that sounds awesome. Thanks for the info. Will check it out. :)
Best regards
Fab

**Comment by Enjoy-Combi** [comment]
Hello,

I have got devices R7060 and having the same issue related the battery and voltage data. The zigbee2mqtt version is 1.25.0-1 managed by homeassistant.

Does-it work for you ?

**Reply by Nerdiyde** [reply]
Hi @Enjoy-Combi,
after updating zigbee2mqtt the bat value is at least displayed in the dashboard. So far I couldnt test it more since I have no access to the setup lately, sorry. :/

Bets regards
Fab

**Reply by Enjoy-Combi** [reply]
Hello,

thanks for the return. On my side I have a "?" for the battery level.

**Reply by Termentum** [reply]
I have the same version of zigbee2mqqt running and also experiencing this :) i can switch the device on and off but there are no voltage and battery values.

Is there anything we can do to help you resolve this issue ?

**Reply by 537719-C** [reply]
Same problem until this morning... I followed what @RenePhilipse Saïd and After asking for data in dev console, I got battery level and percent remaining in zigbee2mqtt devices.

**Comment by jjaviercs** [comment]
Hi, I have the same issue withe Woox R7060 and version Zigbee2MQTT 1.26.0.

In this case, the state of the battery is very important so as not to stay alert and be able to change the batteries on time, especially during vacation periods when you are away from home for a long time.

Please it would be appreciated if someone could give some hint on how to solve it

Regards

**Comment by arteck** [comment]
here also same Problem

https://github.com/ioBroker/ioBroker.zigbee/issues/1498

**Reply by passionsfrucht** [reply]
Sorry for the thumbs down. But it is really annoying to check what's new and then it's just a comment of someone's having the same problem.

You can subscribe to the issue and then get a notification if something changes. Doing this often enough you'll see the problem.

Cheers
